{{cookiecutter.project_slug}}/.pre-commit-config.yaml

```diff
@@ -16,14 +16,14 @@ repos:
       - id: check-case-conflict
       - id: check-docstring-first
       - id: detect-private-key
-
+{%- if cookiecutter.frontend_pipeline != "Webpack" or cookiecutter.frontend_pipeline != "Gulp" or cookiecutter.use_pycharm %}
```

Comment:
I don't understand. What's the problem with having this pre-commit hook? Prettier also formats YAML, markdown and JSON, which may be present in your repos even if you don't use a node based frontend. I use PyCharm too and I still find value in including this...

Reply:
I don't have node globally (I use nvm), and even if I did use it, my frontend repo is usually separate from Django repo, so node wouldn't be in my PATH. Initially, I didn't really understand why pre commit was failing till I saw it try to activate node, so that was the primary reason for opening this: confusion.

But I guess i'm stuck in 2012, cuz I still use bootstrap mostly thanks to cookiecutter-django haha

Reply:
I also use nvm and I don't have any issues with this. I do however have a global node, coming from Homebrew, which now that I think about it, might be a problem with [node based pre-commit](https://pre-commit.com/#node) hooks 🤔  Or maybe not. 


I understand better where the friction comes from regarding Webpack and Gulp, however I don't see how it relates to the usage of PyCharm...

Reply:
Did you mean this?

```suggestion
{%- if cookiecutter.frontend_pipeline in ["Webpack", "Gulp"] %}
```

Reply:
it looked like a vscode only thing given the name, but if not, then for sure can remove

Reply:
Yes, not sure why the tags include vscode in the name, they are not even present anymore in the upstream repo (the pre-commit repo is just a mirror).